src/sync.c

```diff
@@ -100,27 +100,7 @@ bool
 mpd_sync_send_command_v(struct mpd_async *async, const struct timeval *tv0,
```

Comment:
PS: struct timeval is not used here because `mpd_sync_io' is not called anymore;
i have left it since all other functions of mpd_sync receive the timeval as well.
If you prefer that i remove it, let me know.

Reply:
Since this is an internal API, we can remove the parameter easily.
But do we want this? Does this function really need dynamic buffer sizes? What is your reason to refactor the output buffer to be dynamic?

Reply:
The refactoring is to allow  libmpdclient to be limited in its message size only by MPD. If i'm right, currently MPD supports up to 8KiB messages.

Reply:
You didn't answer my question.

Reply:
The reasons were: (i) At least one user would like to send messages > 4KiB (issue #51), and (ii) this code would also adapt in case MPD changes the limit of received messages.
You could also just bump the data size to 8KiB and call it a day (that is possible with that other PR, although, today, the limit is still 4KiB)
Since with this patch we can make room for the message, i don't see the need anymore for `mpd_sync_io`; so i would remove the `tv0` parameter.

Reply:
#51 does not request being able to send large requests. He complains about the lousy error handling if he tries to send a huge malformed request.

Reply:
From the debian bug report:
"Ideally, libmpdclient should support requests of arbitrary size (eventually 
reaching the server limit), but if that is not feasible, at least a proper 
error reporting would be nice."

I think for the majority of cases 4KiB is more than sufficient; The reason i submitted the other PR is in case the changes i'm proposing here are deemed undesirable

Reply:
True, that's where the Debian bug report deviates from #51, but we were talking about #51.
But even the Debian bug report doesn't explain how such a large request might be useful. Just like #51, the Debian bug report describes a malformed request, and supporting malformed requests is not libmpdclient's goal.

Reply:
He also mentions this feature request in #51. The title is about the hard coded limit.

True, the test seems to be about fuzzing, but from the MPD protocol page i didn't find any mention of a limit to the find/search command. In theory, you could concatenate multiple find requests in a single huge line like: `((artist == 'FOO') AND (album == 'BAR')) AND ...`